source/extensions/quic_listeners/quiche/platform/http2_reconstruct_object_impl.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+namespace http2 {
```

Comment:
So, IIUC, you're essentially providing the QUICHE PAL here. This is a bit unfortunate, since it then needs to sit in its own namespace `http2`, which has nothing to do with the rest of the Envoy code base. Should this live in a separate repo?

Reply:
Yes. I guess putting it in a separate repo is one possibility, so long as that does not prevent the files from depending on/referencing Envoy types/libraries. Another possibility would be to add flexibility (e.g. macros?) in QUICHE platform api header files to allow override of the namespace in which the impl types are defined.

Let's chat more about this and the other questions in person--I'll try to find time on your calendar sometime tomorrow (though it looks pretty packed already).